blog/2022-11-21-golang-memory-dump.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Profiling golang in kubernetes
+description: Profiling golang in kubernetes using Kubernetes debug.
+authors: nissessenap
+tags:
+  - pprof
+  - kubernetes
+  - debug
+keywords:
+  - pprof
+  - kubernetes
+  - debug
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+In kubernetes 1.23 the new API ephemeral containers went in to beta and in 1.25 it became stable.
+
+Ephemeral containers or debug containers which is also known as, makes it's possible to attach a container to a already running pod without restarting it.
+
+This is extremely useful when you want to debug your application since they can contain debugging tools that you don't to have in your application container.
```

Comment:
"since the ephemeral container can provide debugging tools that you don't want"

I think I would avoid using "debugging tools" and just say "tools" or similar. If you make a strategy of using ephemeral containers, your production container can contain zero additional binaries - it may not even have a shell.